### ISSUE TITLE
Nuget package FSharp.Compiler.Service v22.0.1 has breaking API change w/o assembly version change

### ISSUE THREAD

**Issue by SteveGilham** [fsprojects/FSharpLint#266]
### Description

Attempting to directly invoke FSharpLint.Core in a .net core environment can give MissingMethodException
```
-> One or more errors occurred. (Method not found: 'Microsoft.FSharp.Compiler.SourceCodeServices.FSharpChecker Microsoft.FSharp.Compiler.SourceCodeServices.FSharpChecker.Create(Microsoft.FSharp.Core.FSharpOption`1<Int32>, Microsoft.FSharp.Core.FSharpOption`1<Boolean>, Microsoft.FSharp.Core.FSharpOption`1<Boolean>, Microsoft.FSharp.Core.FSharpOption`1<Resolver>)'.)
```
This arises because releases of the package up to and including 22.0.1 included assemblies with version 0.0.0.0; previous versions had `Microsoft.FSharp.Compiler.SourceCodeServices.FSharpChecker.Create` take the 4 arguments as in the exception message, but this nuget package version's equivalent has 6 arguments.

Because the assembly version is the same, even though package version 17.0.1 is present in the nuget cache, 22.0.1 may already have been loaded into the current context, or be selected by matching assembly version, to cause this failure.  

From version 22.0.3 on, the assemblies are given separate assembly versions so this ambiguity no longer arises.

### Repro steps

This works for me -- a Fake target containing
```
    !! "**/*.fs"
        |> Seq.iter (fun f -> match Lint.lintFile (Lint.OptionalLintParameters.Default) f (new Version "4.0") with
                              | Lint.Failure x -> new InvalidOperationException(x.ToString()) |> raise
                              | Lint.Success w -> w |> Seq.iter (printfn "%A"))
```
when I have the following compiler service versions present
```
    Directory: C:\Users\steve\.nuget\packages\fsharp.compiler.service


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       19/02/2018     15:55                0.0.82
d-----       05/02/2018     19:58                17.0.1
d-----       24/12/2017     09:21                2.0.0.6
d-----       25/02/2018     11:08                20.0.1
d-----       03/03/2018     09:20                21.0.1
d-----       24/03/2018     10:19                22.0.1
d-----       02/05/2018     08:44                22.0.3
d-----       02/06/2018     15:14                23.0.1
d-----       02/06/2018     15:15                23.0.3
```

### Expected behavior

Lint runs

### Actual behavior

MissingMethodException

### Known workarounds

N/A

Deleting all package versions, and restoring has no effect; Fake 5.1.0 brings in nuget version 23.0.3 only (assembly 23.0.3.0), and with that present, and no 17.0.1 dependency having been restored, the missing method exception still happens. 

### Related information 

* Windows 10
* 0.9.1-beta nuget package
* .net core 2.1.300

**Comment by duckmatt** [comment]
Thanks for the in depth report and solution, I feel like getting to the bottom of this must have been painful! We've since upgraded the version of fcs so I'll close this

**Comment by SteveGilham** [comment]
Can confirm that this was fixed with release 0.10 but I'd simply forgotten this report.

And they said .net was supposed to save us from DLL Hell :)
